### PR TITLE
[cxx-interop] Add Hashable conformance to imported enums

### PR DESCRIPTION
**Pull request by j-hui**
Previously, imported enums only conformed to `RawRepresentable` and `Equatable`,
so they could not be used as members of a `Set` or keys of a `Dictionary`.
This patch adds `Hashable` conformance to give them that ability,
as well as some test cases to clarify the expected behavior.
Existing test cases are updated to reflect this new conformance.

rdar://129713687

### CODE REVIEW

**Comment by j-hui** [comment]
@swift-ci Please test

**Comment by Xazax-hun** [comment]
@swift-ci please smoke test

**Review by Xazax-hun — APPROVED** [review]
LGTM!

**Comment by j-hui** [comment]
@swift-ci Please test

**Comment by j-hui** [comment]
@swift-ci Please test

File: test/Interop/Cxx/enum/hashable-enums.swift [hunk 1]
```diff
@@ -0,0 +1,88 @@
+// REQUIRES: executable_test
```
Comment (egorzhdan) [inline test/Interop/Cxx/enum/hashable-enums.swift:1]: Minor: we usually put `// REQUIRES:` lines after the `// RUN:` lines.

**Review by egorzhdan — COMMENTED** [review]
This is awesome, thank you!

Nice to see that there wasn't any AST generation required for this change. Looks like the Sema logic that derives the `Hashable` conformance kicks in and saves us from the trouble.

I only have a comment about using gyb to auto-generate the test file.

**Comment by j-hui** [comment]
@swift-ci Please smoke test

**Review by egorzhdan — APPROVED** [review]
LGTM!

**Comment by j-hui** [comment]
@swift-ci please smoke test

**Pull request merged**